geoutils/georaster.py

```diff
@@ -1250,41 +1254,62 @@ def interp_points(self,pts,nsize=1,mode='linear',band=1):
         assert mode in ['mean', 'linear', 'cubic', 'quintic',
                         'nearest'], "mode must be mean, linear, cubic, quintic or nearest."
 
-        rpts = []
 
+        rpts = []
         #TODO: might need to check if coordinates are center or point in the metadata here...
 
-
-        xx, yy = self.coords(offset='center', grid=False)
-        #TODO: right now it's a loop... could add multiprocessing parallel loop outside,
-        # but such a method probably exists already within scipy/other interpolation packages?
-        for pt in pts:
-            i,j = self.xy2ij(pt[0],pt[1])
-            if self.outside_image(i,j, index=True):
-                rpts.append(np.nan)
-                continue
-            else:
-                x = xx[j - nsize:j + nsize + 1]
-                y = yy[i - nsize:i + nsize + 1]
-
-                #TODO: read only that window?
-                z = self.data[band-1, i - nsize:i + nsize + 1, j - nsize:j + nsize + 1]
-                if mode in ['linear', 'cubic', 'quintic', 'nearest']:
-                    X, Y = np.meshgrid(x, y)
-                    try:
-                        zint = griddata((X.flatten(), Y.flatten()), z.flatten(), list(pt), method=mode)[0]
-                    except:
-                        #TODO: currently fails when dealing with the edges
-                        print('Interpolation failed for:')
-                        print(pt)
-                        print(i,j)
-                        print(x)
-                        print(y)
-                        print(z)
-                        zint = np.nan
-                else:
-                    zint = np.nanmean(z.flatten())
-                rpts.append(zint)
-        rpts = np.array(rpts)
+        # get coordinates
+        x, y = list(zip(*pts))
+        # if those are in latlon, convert to Raster crs
+        if input_latlon:
+            init_crs = pyproj.CRS(4326)
+            dest_crs = pyproj.CRS(self.crs)
+            transformer = pyproj.Transformer.from_crs(init_crs,dest_crs)
+            x, y = transformer.transform(x,y)
+
+        i, j = self.ds.index(x, y, op=np.float32)
+        i = np.array(i) - 0.5
+        j = np.array(j) - 0.5
+
+        if i.ndim == 0:
+            i = [i]
+            j = [j]
+
+        ind_invalid = [self.outside_image(j[k],i[k],index=True) for k in range(len(i))]
```

Comment:
Have you tried using `np.vectorize` here?

Something like:
```python

ind_invalid = np.vectorize(lambda x, y: self.outside_image(x, y, index=True))(j, i)
```
Using `np.vectorize()` is often orders of magnitude faster than python loops.

Reply:
Done!